qt/scientific_interfaces/ISISReflectometry/Common/ModelCreationHelper.cpp

```diff
@@ -0,0 +1,368 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ModelCreationHelper.h"
+#include "../../ISISReflectometry/Reduction/Batch.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace ModelCreationHelper {
+
+namespace { // unnamed
+Row makeRowWithOutputNames(std::vector<std::string> const &outputNames) {
+  auto row = Row({}, 0.0, TransmissionRunPair(), RangeInQ(), boost::none,
+                 ReductionOptionsMap(),
+                 ReductionWorkspaces({}, TransmissionRunPair()));
+  row.setOutputNames(outputNames);
+  return row;
+}
+} // namespace
+
+/* Rows */
+
+Row makeEmptyRow() {
+  return Row({}, 0.0, TransmissionRunPair(), RangeInQ(), boost::none,
+             ReductionOptionsMap(),
+             ReductionWorkspaces({}, TransmissionRunPair()));
+}
+
+Row makeRow(double theta) {
+  return Row({}, theta, TransmissionRunPair({"22348", "22349"}), RangeInQ(),
+             boost::none, ReductionOptionsMap(),
+             ReductionWorkspaces({}, TransmissionRunPair()));
+}
+
+Row makeRow(std::string const &run, double theta) {
+  return Row(
+      {run}, theta, TransmissionRunPair({"Trans A", "Trans B"}), RangeInQ(),
+      boost::none, ReductionOptionsMap(),
+      ReductionWorkspaces({run}, TransmissionRunPair({"Trans A", "Trans B"})));
+}
+
+Row makeRow(std::vector<std::string> const &runs, double theta) {
+  return Row(
+      runs, theta, TransmissionRunPair({"Trans A", "Trans B"}), RangeInQ(),
+      boost::none, ReductionOptionsMap(),
+      ReductionWorkspaces(runs, TransmissionRunPair({"Trans A", "Trans B"})));
+}
+
+Row makeCompletedRow() {
+  auto row = Row({}, 0.0, TransmissionRunPair(), RangeInQ(), boost::none,
+                 ReductionOptionsMap(),
+                 ReductionWorkspaces({}, TransmissionRunPair()));
+  row.setSuccess();
+  return row;
+}
+
+Row makeRowWithMainCellsFilled(double theta) {
+  return Row({"12345", "12346"}, theta, TransmissionRunPair("92345", "92346"),
```

Comment:
What is the purpose of this with hard-coded values?

Reply:
Few more cases in this file.

Reply:
This is a test helper that needs to be moved at some point and it is present on the umbrella issue for this.